modules/apps/commerce/commerce-product-content-search-web/src/main/resources/META-INF/resources/js/sort/view.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {createPortletURL, navigate} from 'frontend-js-web';
+
+function changeOrderBy(currentURL, portletDisplayId, label) {
+	const newUrl = createPortletURL(currentURL, {
+		orderByCol: label,
+		p_p_id: portletDisplayId,
+	});
+
+	navigate(newUrl.toString());
+}
+
+export default function ({currentURL, namespace, portletDisplayId}) {
+	const optionsLink = document.querySelectorAll(
+		`.dropdown-item.transition-link[id^=${namespace}]`
+	);
+
+	const commerceSortByButton = document.querySelector('.commerce-order-by');
+
+	const dropdownSortByButton = document.getElementById(
+		`${namespace}commerce-dropdown-order-by`
+	);
+
+	function handleOptionsLink(event, link) {
+		event.preventDefault();
+		changeOrderBy(
+			currentURL,
+			portletDisplayId,
+			link.getAttribute('data-label')
+		);
+	}
+
+	optionsLink.forEach((link) => {
+		link.addEventListener('click', handleOptionsLink(link));
```

Comment:
Be careful, here you're not passing the function as a callback, you are passing what the function returns: `undefined`.

If you check the console log you will also see some errors: the first expected parameter `event` is an event but the function is called with `handleOptionsLink(link)`.

This is the solution
```
link.addEventListener('click', (event) => handleOptionsLink(event, link));
```

Or better:
```
link.addEventListener('click', handleOptionsLink));
```

where:

```
	function handleOptionsLink(event) {
		event.preventDefault();

		changeOrderBy(
			currentURL,
			portletDisplayId,
			event.target.getAttribute('data-label')
		);
	}
```